src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs

```diff
@@ -305,45 +307,62 @@ private void TryOpenImage()
 			{
 				try
 				{
-					if (_openedImage is WriteableBitmap wb)
+					if (imageSource is WriteableBitmap wb)
 					{
 						SetFromWriteableBitmap(wb);
 					}
 					// We want to reset the image when there is no source provided.
-					else if (!_openedImage?.HasSource() ?? true)
+					else if (!imageSource?.HasSource() ?? true)
 					{
 						ResetSource();
 					}
-					else if (_openedImage.ImageData != null)
+					else if (imageSource.ImageData != null)
 					{
-						SetSourceBitmap(_openedImage);
+						SetSourceBitmap(imageSource);
 					}
-					else if (_openedImage.BitmapDrawable != null)
+					else if (imageSource.BitmapDrawable != null)
 					{
-						SetSourceDrawable(_openedImage);
+						SetSourceDrawable(imageSource);
 					}
-					else if (_openedImage.ResourceId.HasValue)
+					else if (imageSource.ResourceId.HasValue)
 					{
-						var dummy = SetSourceResource(_openedImage);
+						var dummy = SetSourceResource(imageSource);
 					}
-					else if (_openedImage.FilePath.HasValue() || _openedImage.WebUri != null || _openedImage.Stream != null)
+					else if (imageSource.FilePath.HasValue() || imageSource.WebUri != null || imageSource.Stream != null)
 					{
-						var dummy = SetSourceUriOrStream(_openedImage);
+						// We can't open the image until we have the proper target size, which is 
+						// being computed after the layout has been completed.
+						// However, if we have already determined that the Image has non-finite bounds (eg because it 
+						// has no set Width/Height and is inside control that permits infinite space), then we will never
+						// be able to set a targetSize and must load the image without one.
+						if (imageSource.UseTargetSize && _targetWidth == null && _targetHeight == null && (_hasFiniteBounds ?? true) && !MustOpenImageToMeasure())
```

Comment:
I'm not sure to understand why this applies only to local image?

Reply:
The `_targetHeight` and `_targetWidth` are used only in this case.

Reply:
Why?